lib/hsa/mcwamp_hsa.cpp

```diff
@@ -763,11 +767,56 @@ pool_iterator::pool_iterator()
 ///
 namespace Kalmar {
 
+
+
+// Small wrapper around the hsa hardware queue (ie returned from hsa_queue_create(...).
+// This allows us to see which accelerator_view owns the hsa queue, and
+// also tracks the state of the cu mask, profiling, priority of the HW queue.
+struct RocrQueue {
+    RocrQueue(hsa_queue_t *hwQueue, HSAQueue *hccQueue) : 
+        _hwQueue(hwQueue), _hccQueue(hccQueue) 
+    {
+        assignHccQueue(hccQueue);
```

Comment:
Can we move the hsa_queue creation into the RocrQueue's constructor rather than passing it in as an argument?  I think this is more consistent with the ~RocrQueue where the hsa_queue gets destroyed.